nzgmdb/calculation/distances.py

```diff
@@ -671,6 +706,107 @@ def compute_distances_for_event(
     return propagation_data_combo, extra_event_data
 
 
+def perpendicular_height(
+    point: np.ndarray, base_start: np.ndarray, base_end: np.ndarray
+) -> float:
+    """
+    Compute perpendicular height from a point to a line defined by two points.
+
+    Parameters
+    ----------
+    point : np.ndarray, shape (3,)
+        The point from which the height is measured.
+    base_start : np.ndarray, shape (3,)
+        The start point of the line segment.
+    base_end : np.ndarray, shape (3,)
+        The end point of the line segment.
+
+    Returns
+    -------
+    float
+        The perpendicular height from the point to the line segment.
+    """
+    base_vec = base_end - base_start
+    point_vec = point - base_start
+    cross = np.cross(base_vec, point_vec)
+    base_len = np.linalg.norm(base_vec)
+    return np.linalg.norm(cross) / base_len if base_len else 0.0
+
+
+def inverse_square_integral(
+    sites: np.ndarray, p1: np.ndarray, p2: np.ndarray
+) -> np.ndarray:
+    """
+    Vectorized inverse square integral over a segment for multiple sites.
+
+    Parameters
+    ----------
+    sites : np.ndarray, shape (n_sites, 3)
+        The sites (x, y, z) coordinates.
+    p1 : np.ndarray, shape (3,)
+        The first point of the segment (x, y, z).
+    p2 : np.ndarray, shape (3,)
+        The second point of the segment (x, y, z).
+
+    Returns
+    -------
+    np.ndarray, shape (n_sites,)
+        The integral values for each site.
+    """
+    vec1 = p1 - sites
+    vec2 = p2 - p1
+    A = np.sum(vec1**2, axis=1)
+    B = np.sum(vec1 * vec2, axis=1)
+    C = np.dot(vec2, vec2)
+    D = A * C - B**2
+    # Handle invalid values safely
+    with np.errstate(divide="ignore", invalid="ignore"):
+        sqrt_D = np.sqrt(D)
+        atan_diff = np.arctan((C + B) / sqrt_D) - np.arctan(B / sqrt_D)
+        result = np.where(D > 0, atan_diff / sqrt_D, 0.0)
+    return result
+
+
+def compute_ravg_distance_vectorized(
+    seg_corners: np.ndarray, sites: np.ndarray
+) -> np.ndarray:
+    """
+    Vectorized Ravg calculation for multiple site locations.
+
+    Parameters
+    ----------
+    seg_corners : np.ndarray, shape (3, 4, n_segments)
+        Fault plane corner segments
+    sites : np.ndarray, shape (n_sites, 3)
+        Site (x, y, z) coordinates
+
+    Returns
+    -------
+    np.ndarray
+        Ravg values for each site, shape (n_sites,)
+    """
+    vertical_step = 0.1
+    n_sites = sites.shape[0]
+    n_segments = seg_corners.shape[2]
+    # Set the site depth to 0
+    sites[:, 2] = 0.0
+    sum_inv_rsq = np.zeros(n_sites)
+    for i in range(n_segments):
+        TL, TR, BR, BL = seg_corners[:, :, i].T
+        height = perpendicular_height(TL, TR, BR)
+        n_steps = int(height / vertical_step) + 1
```

Comment:
equivalent to `np.ceil`?
```suggestion
        n_steps = np.ceil(height / vertical_step)
```